src/kv/untyped_map.h

```diff
@@ -809,10 +821,16 @@ namespace kv::untyped
       {
         if (current->version <= version)
         {
+          kv::untyped::Write writes;
+          if (track_deletes_on_missing_keys)
+          {
+            writes = current->writes;
+          }
           changes = std::make_unique<untyped::ChangeSet>(
             roll.rollback_counter,
             current->state,
             roll.commits->get_head()->state,
+            writes,
```

Comment:
Unless I am missing something, this augments the changeset with all the writes, which mostly duplicate current->state except for deletes. Is that correct? If so, and if this is done for expediency, can we have a ticket that tracks that it's possible to optimise this in the future by adding only deletes here?

Reply:
That sounds correct, yes. I'm sure there is some extra duplication going on, will make a ticket.

Reply:
#4483